common/ledger/blkstorage/blockstore_provider.go

```diff
@@ -95,7 +103,20 @@ func NewProvider(conf *Conf, indexConfig *IndexConfig, metricsProvider metrics.P
 // This method should be invoked only once for a particular ledgerid
 func (p *BlockStoreProvider) Open(ledgerid string) (*BlockStore, error) {
 	indexStoreHandle := p.leveldbProvider.GetDBHandle(ledgerid)
-	return newBlockStore(ledgerid, p.conf, p.indexConfig, indexStoreHandle, p.stats), nil
+	return newBlockStore(ledgerid, p.conf, p.indexConfig, indexStoreHandle, p.stats)
+}
+
+// BootstrapFromSnapshottedTxIDs initializes blockstore from a previously generated snapshot
+// Any failure during bootstrapping the blockstore may leave the partial loaded data
+// on disk. The consumer, such as peer is expected to keep track of failures and cleanup the
+// data explicitly.
+func (p *BlockStoreProvider) BootstrapFromSnapshottedTxIDs(
```

Comment:
This is common code, we have not thought yet about what would be the need of orderer if we extend the snapshotting to the orderer - but surely, it won't export TxIDs. So, a general name `snapshot` at this code level may not seem, fit. I cannot think of a reason why it will be cause confusion in the kvledger, as open and bootstrap code path will be entirely different. Still, if we feel that when we reach kvledger level code, changing this would be very simple.

Reply:
`Open(ledgerID string)` and `OpenUsingSnapshot(ledgerID string, snapshot *Snapshot)` is better compared to `Open(ledgerID string)` and `BootstrapFromSnapshottedTxIDs(snapshot *Snapshot)`. By confusion, I was referring that the Open() conveys a clear message but BootstrapFromSnapshottedTxIDs didn't. For example, it is clear whether it opens a new one or bootstrap an already opened one. Maybe I am doing too much nit picking 😄

Reply:
On the naming, the function name should reveal what it does. `Open` means open (no processing involved).